packages/common/core-utils/src/layerCompat.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Result of a layer compatibility check.
+ * @internal
+ */
+export type LayerCompatCheckResult =
+	| { readonly compatible: true }
+	| {
+			readonly compatible: false;
+			readonly generationCompatible: boolean;
+			readonly unsupportedFeatures: string[];
```

Comment:
Add a description specifying that the unsupportedFeatures will be populated only if the generation of the layers was found compatible, also, consider making this variable as optional.